ts/session/apis/snode_api/swarmPolling.ts

```diff
@@ -105,6 +113,33 @@ function entryToKey(entry: GroupPollingEntry) {
   return entry.pubkey.key;
 }
 
+function mergeMultipleRetrieveResults(
+  results: RetrieveMessagesResultsBatched
+): RetrieveMessagesResultsMergedBatched {
+  const mapped: Map<SnodeNamespaces, Map<string, RetrieveMessageItem>> = new Map();
```

Comment:
We usually avoid `Map` but I guess it is the best data structure in this case.

Reply:
We avoid maps because they are not too well supported by ts, but using them as part a function (and not returning one) is fine to me.